docs/conf.py

```diff
@@ -108,15 +108,10 @@
 # -- General configuration ----------------------------------------------------
 
 # If your documentation needs a minimal Sphinx version, state it here.
-needs_sphinx = "3.0"
-
-# The intersphinx_mapping in sphinx_astropy.sphinx refers to astropy for
-# the benefit of other packages who want to refer to objects in the
-# astropy core.  However, we don't want to cyclically reference astropy in its
-# own build so we remove it here.
-del intersphinx_mapping["astropy"]
+needs_sphinx = "8.2.0"  # keep in sync with pyproject.toml
```

Comment:
In principle we could be fancy and import the package metadata and set this dynamically?

Reply:
Ah I thought maybe we could get it from ``importlib.metadata`` as we did for other things lower down

Reply:
ah, actually, the other requirements are *already* using a mix of `importlib.metadata` and `packaging` 😅

Reply:
Actually I'm thinking this might create more maintenance burden then it'll ever save, so I'm reluctant to over-engineer this, see https://github.com/astropy/astropy/issues/17623#issuecomment-3005363198